src/test/java/dev/openfeature/sdk/testutils/FeatureProviderTestUtils.java

```diff
@@ -0,0 +1,30 @@
+package dev.openfeature.sdk.testutils;
+
+import java.time.Duration;
+import java.util.function.Function;
+
+import dev.openfeature.sdk.*;
+import lombok.experimental.UtilityClass;
+
+import static org.awaitility.Awaitility.await;
+
+@UtilityClass
+public class FeatureProviderTestUtils {
+
+    public static void setFeatureProvider(FeatureProvider provider) {
+        OpenFeatureAPI.getInstance().setProvider(provider);
+        waitForProviderInitializationComplete(OpenFeatureAPI::getProvider, provider);
+    }
+
+    private static void waitForProviderInitializationComplete(Function<OpenFeatureAPI, FeatureProvider> extractor, FeatureProvider provider) {
+        await()
+            .pollDelay(Duration.ofMillis(1))
+            .atMost(Duration.ofSeconds(1))
+            .until(() -> extractor.apply(OpenFeatureAPI.getInstance()) == provider);
+    }
```

Comment:
We can replace this with code with code that listens for the READY event when that's complete; that's the coupling between `init/shutdown` and `events` I was talking about - it gives authors a better means of doing this sort of thing without polling, but this works just fine as well.

Reply:
exactly. that's why it is moved out to a dedicated class to just change one place and not all the test cases again.